script.js

```diff
@@ -23,17 +23,32 @@ const addTodoHandler = () => {
     renderTodos(todos);
 };
 
+const deleteTodoHandler = (todoId) => {
+    todos = [...todos].filter((todo) => todo.id !== todoId);
+
+    renderTodos(todos);
+};
+
 const createTodoElement = (todo) => {
     const $todo = document.createElement("li");
+    const $deleteButton = document.createElement("button");
+    $deleteButton.innerText = "Delete";
+
+    eventListenerHandler($deleteButton, "click", () =>
```

Comment:
what does this function do? Is this is factory function? If so let's add a verb to indicate that it does some work.
e.g: **createEventListener**
Also I don't think that we need separate the logic into another function.
```js
$deleteButton.addEventListener("click", () => deleteTodoHandler(todo.id))
```

works fine.

Reply:
Bhai as we have so many eventListeners, For that I created a factory function for the eventListener in utility.js file. Okay I'll remove the function.